### ISSUE TITLE
[Crash] Unfixable conflicts

### ISSUE THREAD

**Issue by Qendolin** [paulevsGitch/BetterEnd#431]
I cannot launch the game with BetterEnd 1.0.0 - 1.0.2.
```
[00:05:13] [main/INFO] (FabricLoader/GameProvider) Loading Minecraft 1.18.1 with Fabric Loader 0.12.12
[00:05:13] [main/INFO] (FabricLoader) Loading 58 mods:
	- bclib 1.3.0
	- betterend 1.0.2
	- cloth-basic-math 0.6.0
	- cloth-config 6.1.48
	- connectiblechains 2.1.0
	- fabric 0.46.4+1.18
	- fabric-api-base 0.4.1+b4f4f6cdc8
	- fabric-api-lookup-api-v1 1.5.0+17be577f65
	- fabric-biome-api-v1 6.0.1+ded849a965
	- fabric-blockrenderlayer-v1 1.1.9+3ac43d9565
	- fabric-command-api-v1 1.1.6+3ac43d9565
	- fabric-commands-v0 0.2.5+b4f4f6cd65
	- fabric-containers-v0 0.1.18+d154e2c665
	- fabric-content-registries-v0 0.4.5+6f53a73d65
	- fabric-crash-report-info-v1 0.1.8+3ac43d9565
	- fabric-data-generation-api-v1 1.1.1+9b1557c165
	- fabric-dimensions-v1 2.1.9+a1d9bbf565
	- fabric-entity-events-v1 1.4.5+6b21378a65
	- fabric-events-interaction-v0 0.4.16+bfa23f1765
	- fabric-events-lifecycle-v0 0.2.6+b4f4f6cd65
	- fabric-game-rule-api-v1 1.0.10+3ac43d9565
	- fabric-gametest-api-v1 1.0.7+c5d03bcd65
	- fabric-item-api-v1 1.3.0+691a79b565
	- fabric-item-groups-v0 0.3.3+3ac43d9565
	- fabric-key-binding-api-v1 1.0.8+c8aba2f365
	- fabric-keybindings-v0 0.2.6+b4f4f6cd65
	- fabric-lifecycle-events-v1 1.4.10+c15ca33565
	- fabric-loot-tables-v1 1.0.8+3ac43d9565
	- fabric-mining-level-api-v1 1.0.3+3ac43d9565
	- fabric-mining-levels-v0 0.1.8+b4f4f6cd65
	- fabric-models-v0 0.3.3+3ac43d9565
	- fabric-networking-api-v1 1.0.18+3ac43d9565
	- fabric-networking-v0 0.3.5+b4f4f6cd65
	- fabric-object-builder-api-v1 1.11.1+3b82842e65
	- fabric-object-builders-v0 0.7.9+3ac43d9565
	- fabric-particles-v1 0.2.9+526dc1ac65
	- fabric-registry-sync-v0 0.9.0+befed49e65
	- fabric-renderer-api-v1 0.4.9+3ac43d9565
	- fabric-renderer-indigo 0.4.12+3ac43d9565
	- fabric-renderer-registries-v1 3.2.7+b4f4f6cd65
	- fabric-rendering-data-attachment-v1 0.3.4+7242e9d765
	- fabric-rendering-fluids-v1 0.1.18+3ac43d9565
	- fabric-rendering-v0 1.1.9+b4f4f6cd65
	- fabric-rendering-v1 1.10.3+6b21378a65
	- fabric-resource-conditions-api-v1 1.0.2+d7c144a865 via fabric
	- fabric-resource-loader-v0 0.4.11+3ac43d9565
	- fabric-screen-api-v1 1.0.7+3ac43d9565
	- fabric-screen-handler-api-v1 1.1.11+3ac43d9565
	- fabric-structure-api-v1 2.0.8+295197a765
	- fabric-tag-extensions-v0 1.2.5+3ac43d9565
	- fabric-textures-v0 1.0.9+3ac43d9565
	- fabric-tool-attribute-api-v1 1.3.5+fb3b57b465
	- fabric-transfer-api-v1 1.5.6+b4f4f6cd65
	- fabricloader 0.12.12
	- java 17
	- minecraft 1.18.1
	- modmenu 3.0.1
	- notenoughcrashes 4.1.3+1.18
Mapping target name conflicts detected:
  METHODs ru/betterend/rituals/InfusionRitual/[net/minecraft/class_2586/method_5431, markDirty]()V -> markDirty
There were unfixable conflicts.
[00:05:18] [main/ERROR] (FabricLoader) Uncaught exception in thread "main"
 java.lang.RuntimeException: Failed to remap mods
	at net.fabricmc.loader.impl.discovery.RuntimeModRemapper.remap(RuntimeModRemapper.java:167) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.setup(FabricLoaderImpl.java:256) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:187) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:142) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:71) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28) ~[fabric-loader-0.12.12.jar:?]
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86) ~[dev-launch-injector-0.2.1+build.8.jar:?]
Caused by: java.lang.RuntimeException: Unfixable conflicts
	at net.fabricmc.tinyremapper.TinyRemapper.handleConflicts(TinyRemapper.java:851) ~[tiny-remapper-0.6.0.jar:0.6.0]
	at net.fabricmc.tinyremapper.TinyRemapper.propagate(TinyRemapper.java:751) ~[tiny-remapper-0.6.0.jar:0.6.0]
	at net.fabricmc.tinyremapper.TinyRemapper.mrjRefresh(TinyRemapper.java:1030) ~[tiny-remapper-0.6.0.jar:0.6.0]
	at net.fabricmc.tinyremapper.TinyRemapper.apply(TinyRemapper.java:882) ~[tiny-remapper-0.6.0.jar:0.6.0]
	at net.fabricmc.loader.impl.discovery.RuntimeModRemapper.remap(RuntimeModRemapper.java:115) ~[fabric-loader-0.12.12.jar:?]
	... 6 more
Disconnected from the target VM, address: '127.0.0.1:55747', transport: 'socket'

Process finished with exit code 1
```

As you can see I only have the required dependencies, connectiblechains and notenoughcrashes loaded. I was also launching the game in from IDEA as I am trying to add compatibility to connectiblechains with your mod.

**Comment by Turachkh** [comment]
Make sure the Bclib version is compatible with the particular betterend version.

**Comment by Qendolin** [comment]
Better End requires version 1.3.0 of bclib which is what I have

**Comment by paulevsGitch** [comment]
> java.lang.RuntimeException: Unfixable conflicts
at net.fabricmc.tinyremapper.TinyRemapper.handleConflicts

Looks like your Fabric installation fails to remap mods, you can try to re-install it

**Comment by Qendolin** [comment]
I'll try but it works with any other mod, even Better Nether

**Comment by Qendolin** [comment]
I've deleted my `bin`, `build`, `.grade` and `run/.fabric` folders and ran 'Clear Caches' in IDEA but I am still getting the same error,

**Comment by Redd56** [comment]
i have the same issue

**Comment by paulevsGitch** [comment]
What mappings are you using?

**Comment by Qendolin** [comment]
yarn 1.18.1+build.22

**Comment by Redd56** [comment]
let me check

**Comment by Redd56** [comment]
same

**Comment by paulevsGitch** [comment]
You can try mojmap, I think this can be a remapper issue due to mappings difference.
You can also try to use mod without remapping (loading in mods folder)

**Comment by Redd56** [comment]
> You can try mojmap, I think this can be a remapper issue due to mappings difference. You can also try to use mod without remapping (loading in mods folder)

thats where the issue arises from, i keep asking on the discord how to use the dev version in ide but so far ive gotten no one to answer me

**Comment by paulevsGitch** [comment]
> how to use the dev version in ide

You can load it as a jar file using gradle, for example

**Comment by Redd56** [comment]
> You can load it as a jar file using gradle, for example

that was via 
flatDir {
		dirs 'libs'
	}

and in dependencys section 
runtimeOnly fileTree('libs') { include '*.jar' }
	with the devs in the libs folder correct?

**Comment by paulevsGitch** [comment]
Yes. this should work

**Comment by Qendolin** [comment]
I'm a little confused right now, I wasn't trying to use BetterEnd as a dependency, I just placed it into my `./run/mods/` folder. Why should I have to add it as a `runtimeOnly` dependency?

**Comment by Redd56** [comment]
The mixins like to crash and burn when being used as a mod in the mods folder due to mapping differences

**Comment by paulevsGitch** [comment]
> No idea why that is, it shouldnt, but they do
Mods from mod folder are remapped during runtime, if mappings are different - they can crash too. I'm not sure what causing this, some mods will work fine, some not

**Comment by Redd56** [comment]
Prolly just some wierdness in mapping matching or missing maping

**Comment by ishland** [comment]
This method needs renaming to fix devlaunch in yarn environment: https://github.com/paulevsGitch/BetterEnd/blob/4e008ab20892e28400cfbc9f856b2a7ce1e36d8b/src/main/java/ru/betterend/rituals/InfusionRitual.java#L231

**Comment by Redd56** [comment]
Markdirty... What a name

**Comment by paulevsGitch** [comment]
Same name as vanilla class uses in other mappings

**Comment by quiqueck** [comment]
Should be fixed in 1.0.3. Please reopen if you are still having issues.
